script.js

```diff
@@ -1,106 +1,112 @@
         listItem.classList.toggle(TASK_COMPLETED_CLASS, this.checked);
 
-        saveTasksToLocalStorage();
-    });
+        saveToLocalStorage();
+    })
 
     deleteButton.addEventListener("click", function () {
-        TODO_LIST.removeChild(listItem);
+        todoList.removeChild(listItem);
 
-        saveTasksToLocalStorage();
-    });
- 
-    const taskNameEditInput = document.createElement("input");
-    taskNameEditInput.type = "text";
-    editButton.addEventListener("click", function () {
+        saveToLocalStorage();
+    })
+
+    const taskEditingInput = document.createElement("input");
+
+    editButtton.addEventListener("click", function() {
         const isEditing = listItem.classList.contains(TASK_EDITING_CLASS);
 
         if (isEditing) {
-            taskNameSpan.textContent = taskNameEditInput.value;
-            listItem.replaceChild(taskNameSpan, taskNameEditInput);
+            taskNameSpan.textContent = this.previousSibling.value;
+            listItem.insertBefore(taskNameSpan, taskEditingInput);
+            taskNameSpan.textContent = taskEditingInput.value;
+            listItem.removeChild(taskEditingInput);
             listItem.classList.remove(TASK_EDITING_CLASS);
-            editButton.textContent = EDIT_BUTTON_LABEL;
+            editButtton.textContent = EDIT_BUTTON;
         } else {
-            taskNameEditInput.value = taskNameSpan.textContent;
-            listItem.replaceChild(taskNameEditInput, taskNameSpan);
+            taskEditingInput.type = "text";
+            taskEditingInput.value = taskNameSpan.textContent;
+            listItem.insertBefore(taskEditingInput, taskNameSpan);
+            listItem.removeChild(taskNameSpan);
             listItem.classList.add(TASK_EDITING_CLASS);
-            editButton.textContent = SAVE_BUTTON_LABEL;
+            editButtton.textContent = SAVE_BUTTON;
         }
-        
-        saveTasksToLocalStorage();
-    });
 
-    saveTasksToLocalStorage();
+        saveToLocalStorage();
+    })
 
+    saveToLocalStorage();
 }
 
-function saveTasksToLocalStorage() {
+function saveToLocalStorage() {
     const tasks = [];
-    document.querySelectorAll("#todo-list li").forEach(task => {
-        const taskName = task.querySelector("span").textContent;
+    document.querySelectorAll("li").forEach((task) => {
+        const taskName = task.firstChild.textContent;
         const isCompleted = task.classList.contains(TASK_COMPLETED_CLASS);
-        tasks.push({ name: taskName, completed: isCompleted });
+        tasks.push({ task: taskName, status: isCompleted });
     });
     localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(tasks));
 }
 
-document.addEventListener("DOMContentLoaded", function () {
-    const savedTasks = JSON.parse(localStorage.getItem(LOCAL_STORAGE_KEY)) ?? [];
+document.addEventListener("DOMContentLoaded", function() {
+    const savedTasks = JSON.parse(localStorage[LOCAL_STORAGE_KEY] ?? []);
```

Comment:
Does `JSON.parse()` expect string as a parameter, not array `[]`?

Reply:
Resolved.